OK for the short term we will have to merge this patch regardless, I swapped to the clang linker but the issue persisted with the link process, even after following the manual.

Also, I think the CI should have detected that these options are broken so I will follow up with a CI change to test ALL sanitizers.
```
>Undefined symbols for architecture x86_64:
  "___ubsan_handle_add_overflow", referenced from:
      List<String, DefaultAllocator>::push_back(String const&) in crash_handler_osx.osx.tools.64s.o
      Vector<Logger*>::push_back(Logger*) in os_osx.osx.tools.64s.o
      List<String, DefaultAllocator>::push_back(String const&) in os_osx.osx.tools.64s.o
      List<String, DefaultAllocator>::push_back(String const&) in display_server_osx.osx.tools.64s.o
      -[GodotContentView insertText:replacementRange:] in display_server_osx.osx.tools.64s.o
      _push_to_key_event_buffer(DisplayServerOSX::KeyEvent const&) in display_server_osx.osx.tools.64s.o
      Vector<String>::push_back(String) in display_server_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_builtin_unreachable", referenced from:
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      OS_Unix::debug_break() in libdrivers.osx.tools.64s.a(os_unix.osx.tools.64s.o)
      _terminator_DestroySurfaceKHR in libdrivers.osx.tools.64s.a(wsi.osx.tools.64s.o)
      _terminator_GetPhysicalDeviceSurfaceSupportKHR in libdrivers.osx.tools.64s.a(wsi.osx.tools.64s.o)
      _terminator_GetPhysicalDeviceSurfaceCapabilitiesKHR in libdrivers.osx.tools.64s.a(wsi.osx.tools.64s.o)
      _terminator_GetPhysicalDeviceSurfaceFormatsKHR in libdrivers.osx.tools.64s.a(wsi.osx.tools.64s.o)
      _terminator_GetPhysicalDeviceSurfacePresentModesKHR in libdrivers.osx.tools.64s.a(wsi.osx.tools.64s.o)
      ...
  "___ubsan_handle_divrem_overflow", referenced from:
      -[GodotContentView firstRectForCharacterRange:actualRange:] in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::mouse_warp_to_position(Vector2i const&) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::screen_get_dpi(int) const in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::_create_window(DisplayServer::WindowMode, Rect2i const&) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::cursor_set_custom_image(Ref<Resource> const&, DisplayServer::CursorShape, Vector2 const&) in display_server_osx.osx.tools.64s.o
      Geometry2D::is_point_in_polygon(Vector2 const&, Vector<Vector2> const&) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::set_icon(Ref<Image> const&) in display_server_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_dynamic_type_cache_miss", referenced from:
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      OS_OSX::initialize_core() in os_osx.osx.tools.64s.o
      OS_OSX::initialize_joypads() in os_osx.osx.tools.64s.o
      OS_OSX::initialize() in os_osx.osx.tools.64s.o
      OS_OSX::finalize() in os_osx.osx.tools.64s.o
      OS_OSX::set_main_loop(MainLoop*) in os_osx.osx.tools.64s.o
      OS_OSX::delete_main_loop() in os_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_float_cast_overflow", referenced from:
      -[GodotWindowDelegate windowDidExitFullScreen:] in display_server_osx.osx.tools.64s.o
      -[GodotWindowDelegate windowDidChangeBackingProperties:] in display_server_osx.osx.tools.64s.o
      -[GodotWindowDelegate windowDidResize:] in display_server_osx.osx.tools.64s.o
      _get_mouse_pos(DisplayServerOSX::WindowData&, CGPoint) in display_server_osx.osx.tools.64s.o
      -[GodotContentView mouseMoved:] in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::mouse_get_absolute_position() const in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::_get_native_screen_position(int) const in display_server_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_function_type_mismatch_v1", referenced from:
      _err_print_error(char const*, char const*, int, char const*, char const*, ErrorHandlerType) in libcore.osx.tools.64s.a(error_macros.osx.tools.64s.o)
      ObjectDB::debug_objects(void (*)(Object*)) in libcore.osx.tools.64s.a(object.osx.tools.64s.o)
      FileAccess::create(FileAccess::AccessType) in libcore.osx.tools.64s.a(file_access.osx.tools.64s.o)
      DisplayServer::vsync_set_enabled(bool) in libservers.osx.tools.64s.a(display_server.osx.tools.64s.o)
      DisplayServer::get_create_function_rendering_drivers(int) in libservers.osx.tools.64s.a(display_server.osx.tools.64s.o)
      DisplayServer::create(int, String const&, DisplayServer::WindowMode, unsigned int, Vector2i const&, Error&) in libservers.osx.tools.64s.a(display_server.osx.tools.64s.o)
      VulkanContext::_obtain_vulkan_version() in libdrivers.osx.tools.64s.a(vulkan_context.osx.tools.64s.o)
      ...
  "___ubsan_handle_implicit_conversion", referenced from:
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      load_address() in crash_handler_osx.osx.tools.64s.o
      CowData<char32_t>::_ref(CowData<char32_t> const&) in crash_handler_osx.osx.tools.64s.o
      CowData<char32_t>::_unref(void*) in crash_handler_osx.osx.tools.64s.o
      unsigned int std::__1::__cxx_atomic_fetch_sub<unsigned int>(std::__1::__cxx_atomic_base_impl<unsigned int>*, unsigned int, std::__1::memory_order) in crash_handler_osx.osx.tools.64s.o
      unsigned int std::__1::__cxx_atomic_load<unsigned int>(std::__1::__cxx_atomic_base_impl<unsigned int> const*, std::__1::memory_order) in crash_handler_osx.osx.tools.64s.o
      bool std::__1::__cxx_atomic_compare_exchange_weak<unsigned int>(std::__1::__cxx_atomic_base_impl<unsigned int>*, unsigned int*, unsigned int, std::__1::memory_order, std::__1::memory_order) in crash_handler_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_invalid_builtin", referenced from:
      _ZSTD_highbit32 in libcore.osx.tools.64s.a(zstd_compress.osx.tools.64s.o)
      _ZSTD_NbCommonBytes in libcore.osx.tools.64s.a(zstd_compress.osx.tools.64s.o)
      _ZSTD_highbit32 in libcore.osx.tools.64s.a(zstd_decompress.osx.tools.64s.o)
      RandomPCG::randf() in libcore.osx.tools.64s.a(random_number_generator.osx.tools.64s.o)
      RandomPCG::randd() in libcore.osx.tools.64s.a(random_pcg.osx.tools.64s.o)
      RandomPCG::randf() in libcore.osx.tools.64s.a(random_pcg.osx.tools.64s.o)
      _BIT_highbit32 in libcore.osx.tools.64s.a(fse_compress.osx.tools.64s.o)
      ...
  "___ubsan_handle_load_invalid_value", referenced from:
      CrashHandler::disable() in crash_handler_osx.osx.tools.64s.o
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      std::__1::__atomic_base<unsigned int, true>::fetch_sub(unsigned int, std::__1::memory_order) in crash_handler_osx.osx.tools.64s.o
      unsigned int std::__1::__cxx_atomic_fetch_sub<unsigned int>(std::__1::__cxx_atomic_base_impl<unsigned int>*, unsigned int, std::__1::memory_order) in crash_handler_osx.osx.tools.64s.o
      std::__1::__atomic_base<unsigned int, false>::load(std::__1::memory_order) const in crash_handler_osx.osx.tools.64s.o
      std::__1::__atomic_base<unsigned int, false>::compare_exchange_weak(unsigned int&, unsigned int, std::__1::memory_order) in crash_handler_osx.osx.tools.64s.o
      unsigned int std::__1::__cxx_atomic_load<unsigned int>(std::__1::__cxx_atomic_base_impl<unsigned int> const*, std::__1::memory_order) in crash_handler_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_missing_return", referenced from:
      CompressEtc2Rgba(unsigned int const*, unsigned long long*, unsigned int, unsigned long) in libmodule_etcpak.osx.tools.64s.a(ProcessRGB.osx.tools.64s.o)
      ProcessAlpha_ETC2(unsigned char const*) in libmodule_etcpak.osx.tools.64s.a(ProcessRGB.osx.tools.64s.o)
      GetMulSel(int) in libmodule_etcpak.osx.tools.64s.a(ProcessRGB.osx.tools.64s.o)
      glslang::TInputScanner::scanVersion(int&, EProfile&, bool&) in libmodule_glslang.osx.tools.64s.a(Scan.osx.tools.64s.o)
      glslang::TScanContext::tokenize(glslang::TPpContext*, glslang::TParserToken&) in libmodule_glslang.osx.tools.64s.a(Scan.osx.tools.64s.o)
      glslang::TIntermediate::findLValueBase(glslang::TIntermTyped const*, bool) in libmodule_glslang.osx.tools.64s.a(Intermediate.osx.tools.64s.o)
      glslang::TPpContext::scanHeaderName(glslang::TPpToken*, char) in libmodule_glslang.osx.tools.64s.a(Pp.osx.tools.64s.o)
      ...
  "___ubsan_handle_mul_overflow", referenced from:
      DisplayServerOSX::screen_get_position(int) const in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::screen_get_usable_rect(int) const in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::_create_window(DisplayServer::WindowMode, Rect2i const&) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::window_get_position(int) const in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::window_set_position(Vector2i const&, int) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::cursor_set_custom_image(Ref<Resource> const&, DisplayServer::CursorShape, Vector2 const&) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::set_icon(Ref<Image> const&) in display_server_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_negate_overflow", referenced from:
      axis_correct(int, int, int) in joypad_osx.osx.tools.64s.o
      AudioServer::_driver_process(int, int*) in libservers.osx.tools.64s.a(audio_server.osx.tools.64s.o)
      String::num_int64(long long, int, bool) in libcore.osx.tools.64s.a(ustring.osx.tools.64s.o)
      String::to_int() const in libcore.osx.tools.64s.a(ustring.osx.tools.64s.o)
      String::to_int(char const*, int) in libcore.osx.tools.64s.a(ustring.osx.tools.64s.o)
      String::to_int(wchar_t const*, int) in libcore.osx.tools.64s.a(ustring.osx.tools.64s.o)
      double built_in_strtod<char>(char const*, char**) in libcore.osx.tools.64s.a(ustring.osx.tools.64s.o)
      ...
  "___ubsan_handle_nonnull_arg", referenced from:
      OpusVorbisDecoder::getPCMS16(WebMFrame&, short*, int&) in libmodule_webm.osx.tools.64s.a(OpusVorbisDecoder.osx.tools.64s.o)
      OpusVorbisDecoder::getPCMF(WebMFrame&, float*, int&) in libmodule_webm.osx.tools.64s.a(OpusVorbisDecoder.osx.tools.64s.o)
  "___ubsan_handle_nullability_arg", referenced from:
      OS_OSX::get_unique_id() const in os_osx.osx.tools.64s.o
      OS_OSX::shell_open(String) in os_osx.osx.tools.64s.o
      OS_OSX::move_to_trash(String const&) in os_osx.osx.tools.64s.o
      -[GodotApplication sendEvent:] in display_server_osx.osx.tools.64s.o
      -[GodotApplicationDelegate forceUnbundledWindowActivationHackStep1] in display_server_osx.osx.tools.64s.o
      -[GodotApplicationDelegate forceUnbundledWindowActivationHackStep2] in display_server_osx.osx.tools.64s.o
      -[GodotApplicationDelegate applicationDidFinishLaunching:] in display_server_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_nullability_return_v1", referenced from:
      -[GodotContentView init] in display_server_osx.osx.tools.64s.o
      -[GodotContentView validAttributesForMarkedText] in display_server_osx.osx.tools.64s.o
      MIDIPacketNext(MIDIPacket const*) in libdrivers.osx.tools.64s.a(midi_driver_coremidi.osx.tools.64s.o)
      dispatch_get_main_queue() in libmodule_camera.osx.tools.64s.a(camera_osx.osx.tools.64s.o)
  "___ubsan_handle_out_of_bounds", referenced from:
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      Variant::clear() in crash_handler_osx.osx.tools.64s.o
      void DirAccess::make_default<DirAccessOSX>(DirAccess::AccessType) in os_osx.osx.tools.64s.o
      remapKey(unsigned int, unsigned int) in display_server_osx.osx.tools.64s.o
      translateKey(unsigned int) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::cursor_set_shape(DisplayServer::CursorShape) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::cursor_set_custom_image(Ref<Resource> const&, DisplayServer::CursorShape, Vector2 const&) in display_server_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_pointer_overflow", referenced from:
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      CowData<char32_t>::_get_refcount() const in crash_handler_osx.osx.tools.64s.o
      Variant::clear() in crash_handler_osx.osx.tools.64s.o
      CowData<char>::_get_refcount() const in crash_handler_osx.osx.tools.64s.o
      CowData<char32_t>::_get_size() const in crash_handler_osx.osx.tools.64s.o
      void DirAccess::make_default<DirAccessOSX>(DirAccess::AccessType) in os_osx.osx.tools.64s.o
      CowData<char32_t>::_get_refcount() const in os_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_shift_out_of_bounds", referenced from:
      next_power_of_2(unsigned int) in os_osx.osx.tools.64s.o
      remapKey(unsigned int, unsigned int) in display_server_osx.osx.tools.64s.o
      sendScrollEvent(int, int, double, int) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::create_sub_window(DisplayServer::WindowMode, unsigned int, Rect2i const&) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::cursor_set_custom_image(Ref<Resource> const&, DisplayServer::CursorShape, Vector2 const&) in display_server_osx.osx.tools.64s.o
      DisplayServerOSX::DisplayServerOSX(String const&, DisplayServer::WindowMode, unsigned int, Vector2i const&, Error&) in display_server_osx.osx.tools.64s.o
      next_power_of_2(unsigned int) in display_server_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_sub_overflow", referenced from:
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      String::length() const in crash_handler_osx.osx.tools.64s.o
      List<String, DefaultAllocator>::_Data::erase(List<String, DefaultAllocator>::Element const*) in crash_handler_osx.osx.tools.64s.o
      Vector<Logger*>::push_back(Logger*) in os_osx.osx.tools.64s.o
      String::length() const in os_osx.osx.tools.64s.o
      Char16String::length() const in display_server_osx.osx.tools.64s.o
      String::length() const in display_server_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_type_mismatch_v1", referenced from:
      CrashHandler::CrashHandler() in crash_handler_osx.osx.tools.64s.o
      CrashHandler::CrashHandler() in crash_handler_osx.osx.tools.64s.o
      CrashHandler::~CrashHandler() in crash_handler_osx.osx.tools.64s.o
      CrashHandler::disable() in crash_handler_osx.osx.tools.64s.o
      CrashHandler::~CrashHandler() in crash_handler_osx.osx.tools.64s.o
      CrashHandler::initialize() in crash_handler_osx.osx.tools.64s.o
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      ...
  "___ubsan_handle_vla_bound_not_positive", referenced from:
      _opus_decode_frame in libmodule_opus.osx.tools.64s.a(opus_decoder.osx.tools.64s.o)
      _opus_decode in libmodule_opus.osx.tools.64s.a(opus_decoder.osx.tools.64s.o)
      _celt_decode_with_ec in libmodule_opus.osx.tools.64s.a(celt_decoder.osx.tools.64s.o)
      _celt_decode_lost in libmodule_opus.osx.tools.64s.a(celt_decoder.osx.tools.64s.o)
      _deemphasis in libmodule_opus.osx.tools.64s.a(celt_decoder.osx.tools.64s.o)
      _celt_synthesis in libmodule_opus.osx.tools.64s.a(celt_decoder.osx.tools.64s.o)
      _silk_Decode in libmodule_opus.osx.tools.64s.a(dec_API.osx.tools.64s.o)
      ...
  "___ubsan_vptr_type_cache", referenced from:
      handle_crash(int) in crash_handler_osx.osx.tools.64s.o
      OS_OSX::initialize_core() in os_osx.osx.tools.64s.o
      OS_OSX::initialize_joypads() in os_osx.osx.tools.64s.o
      OS_OSX::initialize() in os_osx.osx.tools.64s.o
      OS_OSX::finalize() in os_osx.osx.tools.64s.o
      OS_OSX::set_main_loop(MainLoop*) in os_osx.osx.tools.64s.o
      OS_OSX::delete_main_loop() in os_osx.osx.tools.64s.o
      ...
ld: symbol(s) not found for architecture x86_64
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [bin/godot.osx.tools.64s] Error 1
scons: building terminated because of errors.

Process finished with exit code 2
```

--- a/platform/linuxbsd/detect.py
+++ b/platform/linuxbsd/detect.py
@@ -156,7 +156,7 @@ def configure(env):
             if env["use_llvm"]:
                 env.Append(
                     CCFLAGS=[
-                        "-fsanitize=nullability-return,nullability-arg,function,nullability-assign,implicit-integer-sign-change,implicit-signed-integer-truncation,implicit-unsigned-integer-truncation"
+                        "-fsanitize=nullability-return,nullability-arg,function,nullability-assign,implicit-integer-sign-change"
                     ]
                 )
             else:

--- a/platform/osx/detect.py
+++ b/platform/osx/detect.py
@@ -35,7 +35,6 @@ def get_opts():
         BoolVariable("separate_debug_symbols", "Create a separate file containing debugging symbols", False),
         BoolVariable("use_ubsan", "Use LLVM/GCC compiler undefined behavior sanitizer (UBSAN)", False),
         BoolVariable("use_asan", "Use LLVM/GCC compiler address sanitizer (ASAN)", False),
-        BoolVariable("use_lsan", "Use LLVM/GCC compiler leak sanitizer (LSAN)", False),
         BoolVariable("use_tsan", "Use LLVM/GCC compiler thread sanitizer (TSAN)", False),
     ]
 
@@ -132,7 +131,7 @@ def configure(env):
         env["AS"] = basecmd + "as"
         env.Append(CPPDEFINES=["__MACPORTS__"])  # hack to fix libvpx MM256_BROADCASTSI128_SI256 define
 
-    if env["use_ubsan"] or env["use_asan"] or env["use_lsan"] or env["use_tsan"]:
+    if env["use_ubsan"] or env["use_asan"] or env["use_tsan"]:
         env.extra_suffix += "s"
 
         if env["use_ubsan"]:
@@ -142,22 +141,11 @@ def configure(env):
                 ]
             )
             env.Append(LINKFLAGS=["-fsanitize=undefined"])
-            if env["use_llvm"]:
-                env.Append(
-                    CCFLAGS=[
-                        "-fsanitize=nullability-return,nullability-arg,function,nullability-assign,implicit-integer-sign-change,implicit-signed-integer-truncation,implicit-unsigned-integer-truncation"
-                    ]
-                )
-            else:
-                env.Append(CCFLAGS=["-fsanitize=bounds-strict"])
+            env.Append(CCFLAGS=["-fsanitize=nullability-return,nullability-arg,function,nullability-assign"])
 
         if env["use_asan"]:
             env.Append(CCFLAGS=["-fsanitize=address,pointer-subtract,pointer-compare"])
             env.Append(LINKFLAGS=["-fsanitize=address"])
-
-        if env["use_lsan"]:
-            env.Append(CCFLAGS=["-fsanitize=leak"])
-            env.Append(LINKFLAGS=["-fsanitize=leak"])
 
         if env["use_tsan"]:
             env.Append(CCFLAGS=["-fsanitize=thread"])

--- a/platform/server/detect.py
+++ b/platform/server/detect.py
@@ -113,7 +113,7 @@ def configure(env):
             if env["use_llvm"]:
                 env.Append(
                     CCFLAGS=[
-                        "-fsanitize=nullability-return,nullability-arg,function,nullability-assign,implicit-integer-sign-change,implicit-signed-integer-truncation,implicit-unsigned-integer-truncation"
+                        "-fsanitize=nullability-return,nullability-arg,function,nullability-assign,implicit-integer-sign-change"
                     ]
                 )
             else:

